app/controllers/publishers_controller.rb

```diff
@@ -110,14 +110,12 @@ def protect
 
   def ensure_email
     token = params[:token]
-    confirm_email = params[:confirm_email]
-
     publisher = Publisher.find(params[:id])
```

Comment:
I think we have still an issue here: this line should be converted to `publisher = Publisher.find(params[:id]).confirm_pending_email!(confirm_email)` to have `ensure_email` semantic equivalence.

Reply:
@thypon Feel free to go ahead and make whatever changes you need to introduce the functionality you are looking for.  My goal was the fix the 500 impacting users in production.  As I already said, I literally have no idea what the intent of this section of the controller was, I simply knew what was causing the 500 error.  Please enhance the specs I laid out for this method as needed for whatever changes you make.

If you need to confirm an existing email in multiple places it may make the most sense to just create an isolated method that gets called in a callback or within the context of another method.